dev/src/reference.ts

```diff
@@ -1725,22 +1775,33 @@ class QueryUtil<
             docs.reverse();
           }
 
-          resolve(
-            query._createSnapshot(
-              readTime,
-              docs.length,
-              () => docs,
-              () => {
-                const changes: Array<
-                  DocumentChange<AppModelType, DbModelType>
-                > = [];
-                for (let i = 0; i < docs.length; ++i) {
-                  changes.push(new DocumentChange('added', docs[i], -1, i));
-                }
-                return changes;
+          // TODO(ehsannas): Typings of QuerySnapshot suggest that readTime is always available
+          // however there are in the scenario of query plan with no results there will be no read time sent
+          // so the typings should be updated to make readTime optional?
+          // For now we keep the null assertion behaviour which has been the case historically
+          // if (!output.readTime) {
+          //   reject(wrapError(Error('No read time'), stack));
+          //   return;
+          // }
+
+          const result = query._createSnapshot(
```

Comment:
`result` needs to be `null` when `options.analyze !== true`

The backend should only return `explainMetrics` in this case.

However, if `options.analyze === true`, then returning no docs is an empty result, and we should return a snapshot with no docs.